src/io/output_mesh.cc

```diff
@@ -198,6 +198,39 @@ bool OutputMesh::refinement_criterion()
 }
 
 
+void OutputMesh::create_sub_mesh()
+{
+	nodes_.reset();
+	connectivity_.reset();
+	offsets_.reset();
+
+	DebugOut() << "Create output submesh containing only local elements.";
+
+	ElementFullIter ele = ELEMENT_FULL_ITER_NULL(orig_mesh_);
+	int *el_4_loc = orig_mesh_->get_el_4_loc();
+	Distribution *el_ds = orig_mesh_->get_el_ds();
+    const unsigned int n_local_elements = el_ds->lsize();
+
+    orig_element_indices_ = std::make_shared<std::vector<unsigned int>>(n_local_elements);
+
+    offsets_ = std::make_shared<ElementDataCache<unsigned int>>("offsets", (unsigned int)ElementDataCacheBase::N_SCALAR, 1, n_local_elements);
+    Node* node;
+    unsigned int ele_id = 0,
+                 connect_id = 0,
+                 offset = 0,    // offset of node indices of element in node vector
+                 li;            // local node index
+    auto &offset_vec = *( offsets_->get_component_data(0).get() );
```

Comment:
- loop over elements to fill offset and map of orig node IDX to submesh node IDX (use unordered hash map)
- loop over elements to fill connectivity
- loop over map to fill nodes

Reply:
... rather support only discont mesh for parallel submesh, there we can construct mesh in just two loops without using node map.